logstash-core/src/main/java/org/logstash/plugins/internal/Common.java

```diff
@@ -0,0 +1,126 @@
+package org.logstash.plugins.internal;
+
+import java.util.*;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class Common {
+    public static class AddressState {
+        private final String address;
+        private final Set<InternalOutput> outputs = ConcurrentHashMap.newKeySet();
```

Comment:
we get a concurrent hashset , but don't seem to use any of the atomic methods it provides.

Reply:
Yes, but we share it across threads, hence the use of a threadsafe class.

Reply:
This does deserve a comment however.